src/common/AdaptiveStream.cpp

```diff
@@ -1055,22 +1051,24 @@ bool AdaptiveStream::seek_time(double seek_seconds, bool preceeding, bool& needR
   if (newSeg)
   {
     needReset = true;
-    if (newSeg != old_seg)
+    if ((newSeg != old_seg) || (!preceeding && state_ == STOPPED))
```

Comment:
Condition checks of this entire section rewritten to avoid having some parts of the blocks copy/pasted. Please check to see I haven't stuffed it up :)

Reply:
I do not have enough knowledge to verify this code, but i have done various tests and not found side effects,
we will see if there are regressions reported by users